tests/validate.rs

```diff
@@ -212,3 +215,48 @@ async fn test_invalid_header_value() {
     }
     assert!(goose_metrics.errors.len() == 1);
 }
+
+#[tokio::test]
```

Comment:
This should be a in a new file. `tests/validate` is intended to provide test coverage for the [`Validate`](https://docs.rs/goose-eggs/latest/goose_eggs/struct.Validate.html) struct. (Likely a comment should be added to the top of `tests/validate` to make this more clear, as I see that's non-obvious.)